libs/core/src/lib/feed-list-item/directives/line-clamp.directive.ts

```diff
@@ -0,0 +1,207 @@
+import { coerceNumberProperty } from '@angular/cdk/coercion';
+import {
+  Directive,
+  ElementRef,
+  EventEmitter,
+  Input,
+  AfterViewInit,
+  OnDestroy,
+  OnChanges,
+  Output,
+  Renderer2
+} from '@angular/core';
+import { fromEvent, Subscription } from 'rxjs';
+import { debounceTime, distinctUntilChanged } from 'rxjs/operators';
+
+@Directive({
+  selector: '[fdLineClampTarget]',
+  exportAs: 'fdLineClampTarget'
+})
+export class LineClampTargetDirective implements OnChanges, AfterViewInit {
+  @Input()
+  fdLineClampTarget: string;
+  @Output() update = new EventEmitter<LineClampTargetDirective>();
+  constructor(private readonly _elementRef: ElementRef) {}
+
+  get targetElement(): HTMLElement {
+    return this._elementRef.nativeElement;
+  }
+  /** @hidden */
+  ngOnChanges(): void {
+    if (typeof this.targetElement.style.webkitLineClamp === 'undefined') {
+      this.update.emit(this);
+    }
+  }
+  /** @hidden */
+  ngAfterViewInit(): void {
+    this.update.emit(this);
+  }
+}
+
+@Directive({
+  selector: '[fdLineClamp]',
+  exportAs: 'fdLineClamp'
+})
+export class LineClampDirective implements OnChanges, AfterViewInit, OnDestroy {
+  @Input()
+  set fdLineClampLines(value: number) {
+    this._lineCount = coerceNumberProperty(value);
+  }
+  @Input()
+  fdLineClamp = false;
+
+  @Output()
+  lineCountUpdate = new EventEmitter<number>();
+
+  /** @hidden */
+  private _lineClampTarget: HTMLElement;
+  /** @hidden */
+  private _originalText: string;
+  /** @hidden */
+  private windowResize$: Subscription;
+  private _isNativeSupport = true;
+  /** @hidden */
+  private _lineCount: number;
+
+  constructor(
+    private readonly _elementRef: ElementRef,
+    private readonly _renderer: Renderer2
+  ) { }
+
+  get rootElement(): HTMLElement {
+    return this._elementRef.nativeElement;
+  }
+  /** @hidden */
+  ngAfterViewInit(): void {
+    this._isNativeSupport = typeof this.rootElement.style.webkitLineClamp !== 'undefined';
+
+    if (this.rootElement) {
+      this._checkLineCount();
+
+      this.windowResize$ = fromEvent(window, 'resize').pipe(
+        distinctUntilChanged(),
+        debounceTime(200)
+      ).subscribe({
+        next: () => this._checkLineCount()
+      });
+    }
+  }
+  /** @hidden */
+  ngOnDestroy(): void {
+    if (this.windowResize$) {
+      this.windowResize$.unsubscribe();
+    }
+  }
+  /** @hidden */
+  ngOnChanges(): void {
+    this.reset();
+    this.refreshClamp();
+  }
+
+  reset(): void {
+    if (this._lineClampTarget && this._originalText) {
+      this._lineClampTarget.innerText = this._originalText;
+    }
+    if (this._isNativeSupport) {
+      this._resetNative();
+    }
+  }
+
+  refreshTarget(event: LineClampTargetDirective): void {
+    this._lineClampTarget = event.targetElement;
+    this._originalText = event.fdLineClampTarget;
+    this.reset();
+    this.refreshClamp();
+  }
+
+  refreshClamp(): void {
```

Comment:
Please check all your methods, inputs and outputs and add typedocs as they will appear in the public API in the documentation.